openday_scavenger/puzzles/finder/views.py

```diff
@@ -0,0 +1,191 @@
+
+    Args:
+        words (list): word list to generate the puzzle
+
+    Returns:
+        tuple: tuple with puzzle data and solution
+    """
+    # make sure words is a list otherwise return default
+    words = [w for w in words if w]
+    if not words:
+        words = PUZZLE_DEFAULT.split(",")
+
+    return generate_puzzle(words)
+
+
+def get_solution_from_db(puzzle_name: str, db_session: Session) -> list:
+    """
+    Get the puzzle solution from the database session
+    and return it as a list of words.
+
+    Args:
+        puzzle_name (str): puzzle name (entry in the database)
+        db_session (Session): db session
+
+    Returns:
+        list: list of words
+    """
+    try:
+        solution = get(db_session, puzzle_name).answer
+    except Exception as e:
+        solution = PUZZLE_DEFAULT
+        print(f"{e}.\n Puzzle name {puzzle_name} doesn't exist.",
+              f"Using default solution: {solution}")
+    
+    # create word list
+    words = []
+    for word in solution.split(","):
+        words.append(str(word))
+
+    return words
+
+
+@router.get("/static/{path:path}")
+async def get_static_files(
+    path: Path,
+):
+    """Serve files from a local static folder"""
+    # This route is required as the current version of FastAPI doesn't allow
+    # the mounting of folders on APIRouter. This is an open issue:
+    # https://github.com/fastapi/fastapi/discussions/9070
+    parent_path = Path(__file__).resolve().parent / "static"
+    file_path = parent_path / path
+
+    # Make sure the requested path is a file and relative to this path
+    if file_path.is_relative_to(parent_path) and file_path.is_file():
+        return FileResponse(file_path)
+    else:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="Requested file does not exist"
+        )
+
+
+@router.get("/")
+async def index(request: Request,
+                puzzle_name: Annotated[str, Depends(get_puzzle_name)],
+                visitor: Annotated[VisitorAuth, Depends(get_auth_visitor)],
+                db_session: Annotated["Session", Depends(get_db)]
+    ):
+    """
+    Main endpoint for finder puzzle.
+    Gets the word list from the database, fetches the puzzle data,
+    returns the puzzle question and the puzzle data.
+
+    Args:
+        request (Request): Request object
+        puzzle_name (str): puzzle name from database
+        visitor (VisitorAuth): visitor token
+        db_session (Session): db session
+
+    Returns:
+        Response: response object with puzzle data
+    """
+
+    # Get puzzle name, word list, data and metadata
+    solution = get_solution_from_db(puzzle_name, db_session)
+    data, data_as_solution = fetch_puzzle(words=tuple(solution))
+    question = get_quiz(puzzle_name, solution)
+
+    return templates.TemplateResponse(
+        request=request,
+        name="index.html",
+        context={
+            "puzzle": puzzle_name,
+            "visitor": visitor.uid,
```

Comment:
This ("visitor": visitor.uid) is deprecated. This is passed from the cookie now.